### PR TITLE
Exercise 4 Kata. Closes #111

### LINKED ISSUE #111: Add solutions for Exercise4Test in company-kata-solutions
[Exercise4Test](https://github.com/eclipse/eclipse-collections-kata/blob/master/company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java)

### PR DESCRIPTION
**Pull request by GracianoR**
Signed-off-by: Rodrigo Graciano <rodrigo.graciano@bnymellon.com>

### CODE REVIEW

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 1]
```diff
@@ -66,8 +74,8 @@ public void findSupplierNames()
     @Test
     public void countSuppliersWithMoreThanTwoItems()
     {
-        Predicate<Supplier> moreThanTwoItems = null;
-        int suppliersWithMoreThanTwoItems = 0;
+        Predicate<Supplier> moreThanTwoItems = s -> s.getItemNames().length > 2;
+        int suppliersWithMoreThanTwoItems = Lists.mutable.with(this.company.getSuppliers()).count(moreThanTwoItems);
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:78]: Please use `ArrayIterate.count` here.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 2]
```diff
@@ -78,11 +86,13 @@ public void countSuppliersWithMoreThanTwoItems()
     @Test
     public void whoSuppliesSandwichToaster()
     {
+        ImmutableList<Supplier> suppliers = Lists.immutable.with(this.company.getSuppliers());
+
         // Create a Predicate that will check to see if a Supplier supplies a "sandwich toaster".
-        Predicate<Supplier> suppliesToaster = null;
+        Predicate<Supplier> suppliesToaster = s -> Lists.mutable.of(s.getItemNames()).anySatisfy(n -> n.equals(SANDWICH_TOASTER));
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:92]: Is there an `ArrayIterate.contains` method that could be used instead?

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 3]
```diff
@@ -94,9 +104,8 @@ public void whoSuppliesSandwichToaster()
     @Test
     public void filterOrderValues()
     {
-        List<Order> orders = this.company.getMostRecentCustomer().getOrders();
-        MutableList<Double> orderValues = null;
-        MutableList<Double> filtered = null;
+        MutableList<Order> orders = Lists.mutable.ofAll(this.company.getMostRecentCustomer().getOrders());
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:107]: Please use `ListIterate.collect` here instead.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 4]
```diff
@@ -108,9 +117,9 @@ public void filterOrderValues()
     @Test
     public void filterOrderValuesUsingPrimitives()
     {
-        List<Order> orders = this.company.getMostRecentCustomer().getOrders();
-        MutableDoubleList orderValues = null;
-        MutableDoubleList filtered = null;
+        MutableList<Order> orderList = Lists.mutable.ofAll(this.company.getMostRecentCustomer().getOrders());
+        MutableDoubleList filtered = orderList.collectDouble(Order::getValue).select(v -> v > 1.5);
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:121]: Use `ListIterate.collectDouble` here instead.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 5]
```diff
@@ -122,8 +131,8 @@ public void filterOrderValuesUsingPrimitives()
     @Test
     public void filterOrders()
     {
-        List<Order> orders = this.company.getMostRecentCustomer().getOrders();
-        MutableList<Order> filtered = null;
+        MutableList<Order> orderList = Lists.mutable.ofAll(this.company.getMostRecentCustomer().getOrders());
+        MutableList<Order> filtered = orderList.select(o -> o.getValue() > 2.0);
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:135]: Please use `ListIterate.select` here instead.

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 6]
```diff
@@ -94,9 +98,9 @@ public void whoSuppliesSandwichToaster()
     @Test
     public void filterOrderValues()
     {
-        List<Order> orders = this.company.getMostRecentCustomer().getOrders();
-        MutableList<Double> orderValues = null;
-        MutableList<Double> filtered = null;
+
+        MutableList<Double> filtered = ListIterate.collect(
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:102]: Please break this up into two statements with both `orderValues` and `filtered` variables as the test has.

**Comment by GracianoR** [comment]
@donraab, updated as suggested

File: company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java [hunk 7]
```diff
@@ -108,9 +112,8 @@ public void filterOrderValues()
     @Test
     public void filterOrderValuesUsingPrimitives()
     {
-        List<Order> orders = this.company.getMostRecentCustomer().getOrders();
-        MutableDoubleList orderValues = null;
-        MutableDoubleList filtered = null;
+        MutableDoubleList filtered = ListIterate.collectDouble(
```
Comment (donraab) [inline company-kata-solutions/src/test/java/org/eclipse/collections/companykata/Exercise4Test.java:115]: Please break this up into two statements with both `orderValues` and `filtered` variables as the test has.

**Pull request closed without merging**